### ISSUE TITLE
Open modals crash NVDA when exiting

### ISSUE THREAD

**Issue by seanbudd** [nvaccess/nvda#13579]
While `gui.message.messageBox` protects crashes when exiting, other modal dialogs are not protected.
Usage of `ShowModal` and `gui.runScriptModalDialog` cause crashes or undefined behaviour when exiting nvda

### Steps to reproduce:
1. On a web page, open a NVDA find dialog (NVDA+F3)
1. Exit NVDA
1. Note that NVDA doesn't exit and cannot exit in future, it is now in an indeterminate state

---

1. Install a new copy of NVDA
1. Wait for the usage statistics dialog to open
1. Exit NVDA
1. Note that NVDA crashes

---

1. Change the user language of NVDA
1. Wait for the "Restart NVDA for the language change to take affect" dialog to open
1. Exit NVDA
1. Note that NVDA doesn't exit and cannot exit in future, it is now in an indeterminate state

---

1. Go to vision in NVDA preferences, activate screen curtain
1. Wait for the screen curtain warning dialog to open
1. Exit NVDA
1. Note that NVDA doesn't exit and cannot exit in future, it is now in an indeterminate state

### Actual behavior:

NVDA has crashed or partially exited, requiring a manual restart.

### Expected behavior:

Exiting should be blocked until these modals are closed or handled.
A pattern like `messageBox` should be implemented instead.
See #13007 for suggested pattern designs.

#### NVDA installed/portable/running from source:
installed and source

#### NVDA version:
2021.3.4, 2022.1beta3

**Comment by hwf1324** [comment]
The reproduction steps provided so far do not seem to reproduce the issue.

**Comment by CyrilleB79** [comment]
I confirm that the restart seems OK now with the two examples in the STR.

@seanbudd, I think that you have worked again on modal dialogs and restart since this issue has been open. Could you close this issue or clarify what should still be done here if anything?

**Comment by seanbudd** [comment]
I think while there are still instances of `ShowModal` and `gui.runScriptModalDialog` we will run into problems. Did all examples test OK? I imagine there still may be some cases not documented in the issue where using `gui.runScriptModalDialog` and `ShowModal` is causing issues.
However, we could just merge this issue with #13007

**Comment by seanbudd** [comment]
Closing in favour of #13007
